timemachine/lib/potentials.py

```diff
@@ -259,3 +259,16 @@ def unbound_impl(self, precision):
         custom_ctor = getattr(custom_ops, cls_name_base)
 
         return custom_ctor(*self.args)
+
+
+class SummedPotential:
+    def __init__(self, potentials, params):
+        self._impl = custom_ops.SummedPotential(potentials, params)
+
+        # NOTE: the following is necessary to ensure that we maintain
+        # a reference to the wrapped Potential objects on the Python
+        # side; otherwise they may be freed prematurely
+        self._potentials = potentials
```

Comment:
I found I needed this hack to avoid a segfault when calling `execute_device` on the wrapped `SummedPotential`. Without this, I think what's happening is that the wrapped `Potential`s are being freed prematurely (when there are no more references in Python). Curious if we've encountered this issue before and have a standard way of dealing with it?

Reply:
Ahh, good catch.

We have encountered a similar issue before, if you construct a context and don't keep around Python references to its `impl`s. Workarounds like https://github.com/proteneer/timemachine/pull/525#discussion_r740565792 can sometimes be needed.

(In the specific case of contexts though, another workaround is to frequently [construct new contexts](https://github.com/proteneer/timemachine/blob/8b0e4d2281c69b6aba2f62794349fe1e2f7950bf/md/thermostat/moves.py#L14-L27) whenever needed, since context creation overhead in timemachine is quite small.)